src/utils.zig

```diff
@@ -985,10 +985,18 @@ pub fn BlockingMaxHeap(comptime T: type) type {
 }
 
 pub const ThreadPool = struct {
+	pub const TaskType = enum(usize) {
+		chunkgen,
+		lighting,
+		meshgen,
+		misc,
+		taskTypes,
+	};
 	const Task = struct {
 		cachedPriority: f32,
 		self: *anyopaque,
 		vtable: *const VTable,
+		taskType: TaskType,
```

Comment:
Why not make this part of the VTable? There is really no need to store a copy of it in every task.

Reply:
You're right.

Reply:
Oh just saw, that you pushed the commits of the other branch here. I was getting confused here for a moment.